UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs

```diff
@@ -1187,5 +1192,82 @@ private void ColourTile(MatrixInfo matrixInfo, Vector3Int localPos, LayerType la
 		}
 
 		#endregion
+
+		#region DebugCommands
+
+		private static IEnumerator KillLights()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<LightSource>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
+				stationObject.Integrity.ForceDestroy();
+				currentIndex++;
+			}
+		}
+
+		private static IEnumerator SelfPowerEverything()
+		{
+			if (MatrixManager.MainStationMatrix?.Objects == null) yield break;
+
+			var currentIndex = 0;
+			var maximumIndexes = 20;
+			foreach (var stationObject in MatrixManager.MainStationMatrix.Objects.GetComponentsInChildren<APCPoweredDevice>())
+			{
+				if (currentIndex >= maximumIndexes) yield return WaitFor.EndOfFrame;
```

Comment:
Are you wanting to reset `currentIndex` inside this If? or else it will go frame by frame after 20 has been reached for all these coroutines

Reply:
Fixed